projects/driver_controls/src/center_console.c

```diff
@@ -0,0 +1,65 @@
+#include "center_console.h"
+#include "input_event.h"
+// Inputs are as follows:
+// * A0: Power
+// * A1: Drive
+// * A2: Neutral
+// * A3: Reverse
+// * A4: DRLs
+// * A5: Lowbeams
+// * A6: Hazards
+
+static const EventID s_events[NUM_CENTER_CONSOLE_INPUTS] = {
+  INPUT_EVENT_CENTER_CONSOLE_POWER,
+  INPUT_EVENT_CENTER_CONSOLE_DIRECTION_DRIVE,
+  INPUT_EVENT_CENTER_CONSOLE_DIRECTION_NEUTRAL,
+  INPUT_EVENT_CENTER_CONSOLE_DIRECTION_REVERSE,
+  INPUT_EVENT_CENTER_CONSOLE_DRL,
+  INPUT_EVENT_CENTER_CONSOLE_LOWBEAMS,
+  INPUT_EVENT_CENTER_CONSOLE_HAZARDS_PRESSED,
+};
+
+static void prv_hold_timeout(SoftTimerID timer_id, void *context) {
+  event_raise(INPUT_EVENT_CENTER_CONSOLE_POWER, 0);
+}
+
+static void prv_raise_event_cb(GpioExpanderPin pin, GPIOState state, void *context) {
+  CenterConsoleStorage *storage = context;
+
+  switch (pin) {
+    case CENTER_CONSOLE_INPUT_POWER:
+      if (state == GPIO_STATE_LOW) {
```

Comment:
Should also check that the `storage->hold_timer == SOFT_TIMER_INVALID_TIMER` that way if the input is bouncy we avoid starting multiple timers? You'll want to clear the `timer_id` in the callback as well.